app/services/katello/pulp3/repository.rb

```diff
@@ -123,6 +147,40 @@ def distribution_reference(path)
         DistributionReference.find_by(:path => path)
       end
 
+      def refresh_mirror_entities
+        href = mirror_remote_href
+        if href
+          [remotes_api.partial_update(href, mirror_remote_options)]
+        else
+          create_mirror_remote
+          []
+        end
+      end
+
+      def mirror_needs_updates?
+        remote = fetch_remote
+        return true if remote.blank?
+        options = compute_mirror_remote_options
+        options.keys.any? { |key| remote.send(key) != options[key] }
+      end
+
+      def compute_mirror_remote_options
+        computed_options = mirror_remote_options
+        [:ssl_client_certificate, :ssl_client_key, :ssl_ca_certificate].each do |key|
+          computed_options[key] = Digest::SHA256.hexdigest(computed_options[key].chomp)
+        end
+        computed_options
+      end
+
+      def create_mirror_entities
+        create_mirror
+        create_mirror_remote unless mirror_remote_href
+      end
```

Comment:
Do we need this check at creation time?

Reply:
Yes. The idea here is that the case where you have a remote but not a repo also needs to get handled. This line will handle that scenario.